### ISSUE TITLE
Problem with ixc 

### ISSUE THREAD

**Issue by quanganh206** [jxcore/jxcore-cordova#116]
Hi @ktrzeciaknubisa,

I have problem with jxc when try it in mac. 

```
Extracting plugin package...
Command failed: `"/usr/local/bin/jx" io.jxcore.node.jx`:
Error: Command failed: 
module.js:1967
throw err;
      ^
Error: Package is either corrupted or not compatible with this JX binary. (/Users/anhlq/test/jxcore-cordova/hello/io.jxcore.node.jx)
    at Object.Module._extensions..jx (module.js:1964:9)
    at Module.load (module.js:443:30)
    at Function.Module._load (module.js:407:8)
    at Function.Module.runMain (module.js:846:8)
    at startup (node.js:381:8)
    at node.js:1500:1
=========
jx -v 
v0.10.40
```

**Comment by karaxuna** [comment]
So you are running `jxc install` right? To identify problem, Can you please try these scenarios and let me know results?

Try extracting downloaded package manually:

``` bash
cd /Users/anhlq/test/jxcore-cordova/hello
jx io.jxcore.node.jx
```

Download [io.jxcore.node.jx](https://github.com/jxcore/jxcore-cordova-release/raw/master/0.0.4/io.jxcore.node.jx) and try to extract it with `jx io.jxcore.node.jx`?

**Comment by quanganh206** [comment]
Thanks @karaxuna,

I try and it work with android. But I have problem when trying with iOS platform. I try to build with 

```
cordova emulate ios --target='iPhone-5, 8.4'
```

but not success.

**Comment by karaxuna** [comment]
If you extracted `io.jxcore.node.jx` successfully, try re adding plugin:

``` bash
cordova plugins remove io.jxcore.node
cordova plugins add io.jxcore.node
```

I want to know, did this work for you?

``` bash
cd /Users/anhlq/test/jxcore-cordova/hello
jx io.jxcore.node.jx
```

Or you downloaded plugin manually?
What's the error when running `cordova emulate ios --target='iPhone-5, 8.4'` ?

**Comment by quanganh206** [comment]
- I just download plugin in your link. Or I need to install it by jxc install 0.0.4 ? 
- After remove and re-add it still not working with iOS

**Comment by karaxuna** [comment]
I can't reproduce, I'm working on windows, so If package downloaded by jxc is corrupted, then use the link (you already did it). So, what error does `cordova emulate ios --target='iPhone-5, 8.4'` give?

**Comment by quanganh206** [comment]
Here is the error @karaxuna. I think the main problem is iOS platform. I test with Android it is working okay. Did you know what version of iOS compatible? iOS 8.1, 8.4 or 9.2 etc?

I try to update to cordova iOS@4.0.0 but it still not working. Now it is 3.9.2.

```
The following build commands failed:
    Ld build/emulator/HelloWorld.app/HelloWorld normal i386
(1 failure)
Error code 65 for command: xcodebuild with args: -xcconfig,/Users/anhlq/test/hello3/platforms/ios/cordova/build-debug.xcconfig,-project,HelloWorld.xcodeproj,ARCHS=i386,-target,HelloWorld,-configuration,Debug,-sdk,iphonesimulator,build,VALID_ARCHS=i386,CONFIGURATION_BUILD_DIR=/Users/anhlq/test/hello3/platforms/ios/build/emulator,SHARED_PRECOMPS_DIR=/Users/anhlq/test/hello3/platforms/ios/build/sharedpch
ERROR running one or more of the platforms: Error: /Users/anhlq/test/hello3/platforms/ios/cordova/run: Command failed with exit code 2
```

**Comment by karaxuna** [comment]
Don't know that. Can you try re adding platforms too?

``` bash
cordova platform remove ios
cordova platform remove android
cordova plugins remove io.jxcore.node
cordova plugins add io.jxcore.node
cordova platform add ios
cordova platform add android
```

If it does not help, we'll need @ktrzeciaknubisa to reproduce problem on iOS.

**Comment by quanganh206** [comment]
Thanks @karaxuna, still not working.

Waiting for @ktrzeciaknubisa, I test with other cordova plugins iOS working normally.

Hope that help:

```
Ld build/emulator/HelloWorld.app/HelloWorld normal i386
    cd /Users/anhlq/test/hello1/platforms/ios
    export IPHONEOS_DEPLOYMENT_TARGET=6.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/opt/local/bin:/opt/local/sbin:/Users/anhlq/.rbenv/shims:/Users/anhlq/.rbenv/bin:/Users/anhlq/.rbenv/shims:/Users/anhlq/.rbenv/bin:/Users/anhlq/.rbenv/bin:/Users/anhlq/.rbenv/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch i386 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.2.sdk -L/Users/anhlq/test/hello1/platforms/ios/build/emulator -L/Users/anhlq/test/hello1/platforms/ios/HelloWorld/Plugins/io.jxcore.node -F/Users/anhlq/test/hello1/platforms/ios/build/emulator -filelist /Users/anhlq/test/hello1/platforms/ios/build/HelloWorld.build/Debug-iphonesimulator/HelloWorld.build/Objects-normal/i386/HelloWorld.LinkFileList -mios-simulator-version-min=6.0 -Xlinker -objc_abi_version -Xlinker 2 -weak_framework CoreFoundation -weak_framework UIKit -weak_framework AVFoundation -weak_framework CoreMedia -weak-lSystem -ObjC -fobjc-arc -fobjc-link-runtime -framework AssetsLibrary /Users/anhlq/test/hello1/platforms/ios/build/emulator/libCordova.a -framework CoreGraphics -framework MobileCoreServices -lcares -lchrome_zlib -lhttp_parser -ljx -lmozjs -lopenssl -lsqlite3 -luv -lleveldb -lleveldown -lsnappy -framework SystemConfiguration -lstdc++.6.0.9 -framework CFNetwork -framework Foundation -framework Security -Xlinker -dependency_info -Xlinker /Users/anhlq/test/hello1/platforms/ios/build/HelloWorld.build/Debug-iphonesimulator/HelloWorld.build/Objects-normal/i386/HelloWorld_dependency_info.dat -o /Users/anhlq/test/hello1/platforms/ios/build/emulator/HelloWorld.app/HelloWorld
ld: in /Users/anhlq/test/hello1/platforms/ios/HelloWorld/Plugins/io.jxcore.node/libcares.a(ares_strcasecmp.o), building for iOS simulator, but linking in object file built for OSX, for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by quanganh206** [comment]
Hi @karaxuna & @ktrzeciaknubisa,

I am success build iOS with 0.0.8 yet. Thanks for your support!

**Comment by ktrzeciaknubisa** [comment]
Hi guys. I don't have resources right now to test jxcore-cordova against iOS 8.4. I have just checked it only with iPhone 5, 8.1, but didn't find any issue.

@quanganh206 so finally how did you manage to succeed? With `jxc` or manual steps given by @karaxuna ? Are all problems gone or you still need some assistance?

As a side note, if there is a need to download manually the _io.jxcore.node.jx_ package, we can still tell `jxc` to use it:

``` js
$ jxc install /path/to/io.jxcore.node.jx --force
```

**Comment by quanganh206** [comment]
I did with manual steps. 
- I download 0.0.8 and jx to extract it. Old version not compatible with iOS, always have link error. I will try to test with 0.0.7 and 0.0.5.
- jxc always fail when I try to test from the start. I think it have problem when downloading resource.

My environment:
- **jx**: v.0.10.40
- **cordova**: 5.4.1
- **iOS**: 8.4 

Note: I tested with iOS 9.2, 0.0.8 plugin running okay. Thanks!
